### ISSUE TITLE
Matching engine and trade executor generating errors

### ISSUE THREAD

**Issue by mod** [openware/peatio#1324]
# Bug

Trade-Executor daemon receive outdated trades, order are already empty or in state 'done'
also when this order is invalid it's sending it back to matching process which is pointless.

### Reproduce
To reproduce you need to run stress tests or our locus benchmark
By creating a lot of order we notice that the queue of trade.new never gets empty.

There is a lot of error from *trade-executor* output.

We notice the following errors:
- Volume is empty
- State is done

## Suggestions

We need to document every error with a Worker error code
Error need to serialize in JSON with message + error code

When error occurs we need to Queue them into an Error Queue
with the debug payload.

Error should not be send back to matching, eventually we can have a retry-max strategy.

## Error Consumer

We can commit a job CLI to consume counter and dump errors from the queue.
Maybe one day display them in Admin panel.

## Other problems

Base on an error diagnostic patch we need to understand why 70% of trade-executor message are outdated and duplicate trade (already done or already fulfilled)

**Comment by mod** [comment]
https://github.com/rubykube/peatio/blob/1-8-stable/app/trading/matching/executor.rb#L39
```
    def valid?
      return false if @ask.ord_type == 'limit' && @ask.price > @price
      return false if @bid.ord_type == 'limit' && @bid.price < @price
      return false unless @ask.state == Order::WAIT
      return false unless @bid.state == Order::WAIT
      @funds > ZERO && [@ask.volume, @bid.volume].min >= @volume
    end
```

We will need 1 Error type per type of error
We need a function validate() which raise as many expection as error type we have to debug.

Once we collect errors we need to queue them in a AMQP Queue.
https://github.com/rubykube/peatio/blob/1-8-stable/app/trading/matching/executor.rb#L23
```
AMQPQueue.enqueue(:failure, error: error.to_json)
```

**Comment by dinesh-skyach** [comment]
@mod to reproduce this issue do I need to use https://github.com/rubykube/toolbox ?

**Comment by pkdcryptos** [comment]
Trading worker will never submit to matching process again. These trades are with volume  of almost close to 0, but not zero. These trades happen with stress testing, because you will see orders filled by 99.9999% in first match and the rest of 0.0001% get filled by next counter order.

**Comment by yivo** [comment]
Closed via #1328.
